### PR TITLE
Chore: GitHub workflow multiple OS and general updates

### PR DESCRIPTION
**Pull request by airvzxf**
It is a big refactor of the workflows in GitHub based on the issue https://github.com/Col-E/InstrumentationServer/issues/10.

**Note:**
I can't test the release jobs in the new workflow, but they should be work because I didn't touch the main logic, only the conditions. For any inconvenience, please let me know or contact me directly at my email.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,85 +1,175 @@
...
-  build:
+  latest-java-version:
+    name: Get latest Java version
+    runs-on: ubuntu-latest
+    steps:
+      - name: API Adoptium - Get Java versions
+        run: |
+          LATEST_JAVA_VERSION=$(curl --silent \
+              -X 'GET' \
+              'https://api.adoptium.net/v3/assets/version/%5B1.0%2C111.0%5D?heap_size=normal&image_type=jdk&page=0&page_size=1&project=jdk&release_type=ga&sort_method=DEFAULT&sort_order=DESC&vendor=eclipse' \
```
Comment (Col-E) [inline .github/workflows/build.yml]: I think the `info/available_releases` endpoint would be a simpler call: `https://api.adoptium.net/v3/info/available_releases`

Then all you need to do is read the `most_recent_feature_release` field.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,85 +1,175 @@
...
+          path: artifacts
+      - name: Publish test results
+        uses: EnricoMi/publish-unit-test-result-action@v2
+        with:
+          check_name: Unit Test results
+          files: artifacts/unit-test-*/**/*.xml
+      - name: Display files
+        run: ls -lha .
+      - name: Display structure of downloaded files
+        run: ls -lhaR artifacts
```
Comment (Col-E) [inline .github/workflows/build.yml]: I think these `ls` commands can be removed or commented out now that the CI goals have been accomplished

Comment (airvzxf) [reply .github/workflows/build.yml]: Yes, I forgot to delete these steps. Removed in commit [7be7c67](https://github.com/Col-E/InstrumentationServer/pull/11/commits/7be7c678bc75becb037fab4edcaf1430c8d4b9f5)

File: .github/workflows/build.yml (hunk 1 above)
Comment (airvzxf) [reply .github/workflows/build.yml]: Awesome! I applied the change in the commit [d551f0a](https://github.com/Col-E/InstrumentationServer/pull/11/commits/d551f0ab09928f1e5aa692bd17a54b65c7146c5d).

**Comment by Col-E** [comment]
If you want to consolidate the commits and provide a summary you can do something like:

`git rebase -i HEAD~103`

Mark the commits you wanna collapse with `f` then the kept commit with `r` to reword the commit summary. One git push force later, the commit history looks much more organized. I probably could have done that with my old CI work earlier, but oh well 😅

**Comment by airvzxf** [comment]
> If you want to consolidate the commits and provide a summary you can do something like:
> 
> `git rebase -i HEAD~103`
> 
> Mark the commits you wanna collapse with `f` then the kept commit with `r` to reword the commit summary. One git push force later, the commit history looks much more organized. I probably could have done that with my old CI work earlier, but oh well sweat_smile

I always have this problem with the commits. I mean, usually I have many commits, and then they need to be compressed. There is another option, and I don't know if you want to try it, which is the `squash commits`. Basically, it applies one commit with description when the merge is done. In your case, after the merge approval, it will show in master branch one commit with your description. Furthermore, it offers an advantage, you can add the description of the commit, following your own rules and desires. Otherwise, please let me know, and I will apply the rebase solution:  [This solution on StackOverflow shows the steps to squash commits manually](https://stackoverflow.com/questions/5189560/how-do-i-squash-my-last-n-commits-together/61171280#61171280).

---

Avoiding the manual process that I need to do, which is reset the commits and commit again after every commit in my local branch, it is a solution which is directly squash commits in the pull request.

[GitHub Docs: Configuring commit squashing for pull requests](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests).

GitHub-squash-commits-from-pull-request

**Comment by Col-E** [comment]
I was initially going to do a squash, but then thought to double check if you wanted to control what went into that first.

**Comment by airvzxf** [comment]
> I was initially going to do a squash, but then thought to double check if you wanted to control what went into that first.

No problem, please go ahead.

**Comment by Col-E** [comment]
Thank you again for the awesome work!

**Pull request merged**